### ISSUE TITLE
stringByAppendingFormat:

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#144]
```
title = [title stringByAppendingFormat:@" - %0.2f", [[self amount] doubleValue]];
```

Result:

```
title = title + " - %0.2f"
```

Notice that `%0.2f` is left there and the argument is discarded. This is likely related to the other issues with formatting. Here is an idea how to make this safer for now, since this is incredibly dangerous behavior that can mess up your code if you do not double check every converted line:

- if you count more arguments of a formatting function than declared formatting options within the string, just leave the string and use `String(format: "formatting string", ...)`. So in this case, since `%0.2f` isn't supported by either iSwift, or natively by Swift itself, the result would look like: `title = title + String(format: " - %0.2f", self.amount.doubleValue)`.

**Comment by drkameleon** [comment]
Basically, the issue is more... generalised (and not just `stringByAppendingFormat:`-specific).

Actually, it affects every format string where float precision is involved (see: http://stackoverflow.com/questions/24051314/precision-string-format-specifier-in-swift).

I'm looking into right now and I believe all of the case can be solved... Let's see...

**Comment by charlieMonroe** [comment]
See this comment - that's IMHO the best way. http://stackoverflow.com/a/24102844/1800936

**Comment by drkameleon** [comment]
That's exactly what I'm doing... right now

Basically the logic is:

- If it's a "normal" format string (without precision specifiers), just do it the "normal" way with string interpolation
- If the format string contains precision specifiers, then use a `String(format: XXX, args)` construct

I'll post an example once I get it working. Lots of updates coming - very soon. Stay tuned! ;-)

**Comment by drkameleon** [comment]
Issue fixed as of the upcoming 2.3.
